src/ContentType/Request/SmoothRedirectRequestHandler.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of sensiolabs-de/storyblok-bundle.
+ *
+ * (c) SensioLabs Deutschland <info@sensiolabs.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Storyblok\Bundle\ContentType\Request;
+
+use Storyblok\Api\Domain\Value\Dto\Version;
+use Storyblok\Api\Request\StoryRequest;
+use Storyblok\Api\Response\StoryResponse;
+use Storyblok\Api\StoriesApiInterface;
+use Storyblok\Bundle\Routing\Route;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Contracts\HttpClient\Exception\ClientExceptionInterface;
+
+/**
+ * When a slug was not found it tries to find the parent slug and redirects to it. When this fails it throws a
+ * HttpNotFoundException.
+ */
+final readonly class SmoothRedirectRequestHandler implements RequestHandlerInterface
+{
+    public function __construct(
+        private StoriesApiInterface $stories,
+        private UrlGeneratorInterface $urlGenerator,
+        private string $version,
+    ) {
+    }
+
+    public function handle(Request $request, string $slug): Response|StoryResponse
```

Comment:
I'm not convinced by this approach, at least I would change its name to something more explicit (redirect to parent...).

Even if it's disabled by default, it just requires a misconfiguration and it could be hard to detect what's wrong. Maybe requiring a more explicit way to enabling it?

Reply:
we have now a differnt approach. i still like to enable it via config because it is helps with SEO optimization out of the box